surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDbg
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
+Type: select all that apply
+
+- Editor extension (e.g. `termdebug`)
+- Built-in IDE debugger
+- Terminal line interface (e.g. `gdb`)
+- Terminal UI (e.g. `gdb`'s TUI)
+- Web interface (e.g. Pernosco)
+- Separate debugging program
+
+> *Justification:*
+>
+> Interface used for a debugger materially changes the experience of using a
+> debugger and the advantages/disadvantages of it.
+
+### What pain points have you experienced using a debugger with Rust?
+Type: select all that apply
+
+- Often step too many times and need to restart from scratch
+- Don't have a source view of the current line being run
+- Unintuitive interface/errors
+- Expression parser doesn't support expressions that you want to write
+- Optimized-out code
+- Conditional breakpoints don't trigger
+- Difficult to trace dataflow
+- Too much detail/irrelevant information (i.e. assembly views)
+- Incorrect information displayed (i.e. line numbers, variable values)
+- Current line number jumps around unexpectedly
+- Values aren't visualized as expected (e.g. `Vec<T>` is shown as a pointer,
+  length and capacity rather than the elements)
+
+> *Justification:*
+>
+> It might be possible for us to address these - or they might be the result of
+> the compiler's debuginfo output.
+
+
+### Does your Rust debugging typically cross language boundaries?
+Type: yes/no
+
+> *Justification:*
+>
+> If lots of users are using debuggers when doing cross-language interop, then
+> support for this could be improved.
+
+### Which programming language(s) do you debug across (with Rust)?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- ...
+
+> *Justification:*
+>
+> Debugging Rust + C is very different from Rust + Python and the quality of
+> the experience is likely very different too.
+
+
+### Have you ever written a visualization for a debugger (with Rust)?
+Type: select all that apply (optional)
+
+- `gdb`
+- `lldb`
+- `NatVis`
+- WinDbg (not `NatVis`)
+- Visual Studio (not `NatVis`)
+
+> *Justification:*
+>
+> Part of high-quality debugger support is things like type visualizations that
+> could be shipped by libraries, knowing what experience Rust users have with
+> writing these visualizations would be valuable.
+
+
+### Have you ever used a debugger on macro-generated code in Rust?
```

Comment:
I don't think these are useful questions - there are so many different levels of macro-generated code that experiences will differ widely, especially with attribute proc macros. Users may not understand whether a problem is due to a macro or not. If you really want to cover this, I would just have a free form question "Please tell us about any issues you've encountered which were probably caused by macros" or similar

Reply:
I'd like to keep some questions that differentiate these - they might not be suitable for all users but for those who can answer them, I think we'd like to know the answers. So I've added a "not sure" option to the existing "yes" and "no", which will just skip the following question like "no" does.